File: uri/5 - Mathematics/2148/Main.java
import java.util.*;

public class Main{
static int s,d;
static long [][] memo = new long[100][100];

static long solve(int sum, int dados){
	if(dados==d)
		return sum == s ? 1 : 0;
	if(sum >= s)
		return 0L;
	if(memo[sum][dados] != -1)
		return memo[sum][dados];

	long ans = 0;
	for(int i=1; i <= 6; i++)
		ans += solve(sum+i, dados+1);

	return memo[sum][dados]=ans;
}
static void memset(long a[][], long value){

	for(int i=0; i < a.length; i++){
		for(int j=0; j < a[i].length; j++){
			a[i][j] = value;
		}
	}

}
public static void  main(String [] args){
	Scanner scanner = new Scanner(System.in);
	int tc = scanner.nextInt();
	while( (tc--) > 0 ){
		s = scanner.nextInt();
		d = scanner.nextInt();
		memset(memo, -1);
		System.out.printf("%.15f\n", solve(0,0) / Math.pow(6,d));
	}

}

}

File: uri/5 - Mathematics/2148/Main.cpp
#include <cstdio>
#include <cstring>
#include <cmath>

typedef long long ll;

int s,d;
ll memo[100][100];

ll solve(int sum, int dados){
	if(dados==d){
		return sum == s;
	}
	if(sum >= s)
		return 0;
	if(memo[sum][dados] != -1)
		return memo[sum][dados];

	ll ans = 0;
	for(int i=1; i<= 6; i++)
		ans += solve(sum+i, dados+1);

	return memo[sum][dados]=ans;
}

int main(){

	int tc;
	scanf("%d", &tc);
	while(tc--){
		scanf("%d %d", &s, &d);
		memset(memo, -1, sizeof memo);
		printf("%.15Lf\n", solve(0,0) / (long double)powl(6,d));
	}
	return 0;
}